File: CookBook/API/Models/Entities/Recipe.cs
namespace API.Models.Entities;

public class Recipe
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public DateTime LastEdit { get; set; }
    public Guid CategoryId { get; set; }
    public virtual Category Category { get; set; } = null!;
    public virtual ICollection<Ingredient> Ingredients { get; set; } = null!;
}

File: CookBook/client-app/src/app/models/recipe.ts


export interface Recipe {
    id: string;
    name: string;
    lastEdit: Date;
    category: string;
}

export interface RecipeById {
    id: string;
    name: string;
    description: string;
    lastEdit: Date;
    category: string;
    ingredients: string[];
}

export interface RecipeFormValues {
    name: string;
    description: string;
    category: string;
    ingredients: string[];
}